src/routines/solve_routines.jl

```diff
@@ -1,37 +1,68 @@
 """ Solves Operational Models"""
-function solve_op_model!(op_model::OperationModel; kwargs...)
 
-    timed_log = Dict{Symbol, Any}()
+function _write_op_model(results::OperationModelResults, save_path::String)
+    
+    if !isdir(save_path) 
 
-    if op_model.canonical.JuMPmodel.moi_backend.state == MOIU.NO_OPTIMIZER
+        @error "Specified path is not valid. Run write_memory_results to save results."
 
-        if !(:optimizer in keys(kwargs))
+    else
 
-            error("No Optimizer has been defined, can't solve the operational problem")
+        new_folder = mkdir("$save_path/$(round(Dates.now(),Dates.Minute))")
+        folder_path = new_folder
+        write_variable_results(results.variables, folder_path) 
+        write_optimizer_results(results.optimizer_log, folder_path)
+        write_time_results(results.times, folder_path)
+        println("Files written to $folder_path folder.")
+    
+    end
 
-        else
-            _, timed_log[:timed_solve_time],
-            timed_log[:solve_bytes_alloc],
-            timed_log[:sec_in_gc] = @timed JuMP.optimize!(op_model.canonical.JuMPmodel,
-                                                          kwargs[:optimizer])
+return results
```

Comment:
There is no need to have a return here. The function should only write to disk the results.